src/runtime/hexagon/hexagon_device_api.h

```diff
@@ -164,6 +176,12 @@ class HexagonDeviceAPI final : public DeviceAPI {
 
   //! \brief Current buffer manager
   HexagonBufferManager* mgr;
+
+  //! \brief Thread manager
+  std::unique_ptr<HexagonThreadManager> runtime_threads;
+  const unsigned threads{6};
```

Comment:
`pipe_size` can probably be increased. This number will determine the max depth of the dispatch queue for each thread, which might fill up with size 100. E.g., if you dispatch all the compute calls for a kernel before starting the threads, it might come out to >100 dispatches for a given thread if the kernel is big enough.

I don't think there's a negative associated with increasing it, so maybe 1,000 words?

Reply:
Thank you - I'll make that update